### PR TITLE
Feat: Add Translate Function And Translate DB contents as a Task

### PR DESCRIPTION
**Pull request by abdullai-t**
####  Summary / Highlights
This pull request introduces several enhancements and bug fixes to the translation functionality within our application. Over the past two days, we have made significant progress in improving the efficiency and reliability of our translation services. Below is a summary of the changes included in this pull request:

- Added the functionality to run through the database and translate contents into supported languages..
- Added a middleware (`TranslationMiddleware`) to process response data and retrieve translations for the dictionary values.
-  Added a new file containing the static texts from the campaign site
-  Added an endpoint to fetch static content in a specific language for the campaign site.
- Added a new task to translate the database contents to all supported languages

#### Requirements (place an `x` in each `[ ]`)
* [x] I've read and understood the [Contributing Guidelines](/blob/main/.github/contributing.md) and have done my best effort to follow them.
* [x] I've tested my changes manually.
* [ ] I've added unit tests to my PR.

##### Transparency ([Project board](https://github.com/orgs/massenergize/projects/7/views/2))
* [x] I've given my PR a meaningful title.
* [x] I linked this PR to the relevant issue.
* [ ] I moved the linked issue from the "Sprint backlog" to "In progress" when I started this.
* [ ] I moved the linked issue to "QA Verification" now that it is ready to merge.

### CODE REVIEW

**Review by Opoku-Agyemang — CHANGES_REQUESTED** [review]
this PR is also tooo big.  Ideally we want to separate things.  For eg. the PR to introduce a new model should be different.  The translation middleware should also be a different PR .

File: src/_main_/utils/stage/__init__.py [hunk 1]
```diff
@@ -185,4 +185,19 @@ def get_trusted_origins(self):
...
+            "project_id": os.environ.get('GOOGLE_TRANSLATE_PROJECT_ID'),
+            "private_key_id": os.environ.get('GOOGLE_TRANSLATE_PRIVATE_KEY_ID'),
+            "private_key": os.environ.get('GOOGLE_TRANSLATE_PRIVATE_KEY'),
+            "client_email": os.environ.get('GOOGLE_TRANSLATE_CLIENT_EMAIL'),
+            "client_id": os.environ.get('GOOGLE_TRANSLATE_CLIENT_ID'),
+            "auth_uri": "https://accounts.google.com/o/oauth2/auth",
+            "token_uri": "https://oauth2.googleapis.com/token",
+            "auth_provider_x509_cert_url": "https://www.googleapis.com/oauth2/v1/certs",
+            "client_x509_cert_url": os.environ.get('GOOGLE_TRANSLATE_CLIENT_X509_CERT_URL'),
+            "universe_domain": os.environ.get('GOOGLE_TRANSLATE_UNIVERSE_DOMAIN')
```
[the remark this answers is not in the document]
Comment (abdullai-t) [reply src/_main_/utils/stage/__init__.py:201]: @archx3 has fixed it he is working on pushing it

File: src/task_queue/database_tasks/translate_contents.py [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+
+from _main_.utils.activity_logger import ActivityLogger
+from _main_.utils.translation import JsonTranslator
+from _main_.utils.utils import generate_text_hash
+from database.models import SupportedLanguage, TextHash, TranslationsCache
+from translation.translator import Translator
+
+SOURCE_LANGUAGE_CODE = 'en'
+
+logger = ActivityLogger()
```
Comment (Opoku-Agyemang) [inline src/task_queue/database_tasks/translate_contents.py:15]: you wanna do 
```suggestion
from _main_.utils.massenergize_logger import log
```

File: src/task_queue/database_tasks/translate_contents.py [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+    def translate_text(self, text, target_language_code):
+        try:
+            if not text:
+                return None
+            translated_text, err = self.translator.translate(text, target_language_code)
+            if err:
+                raise TranslationError(f"Error translating text: {err}")
+            return translated_text
+        except Exception as e:
+            logger.log(f"Error occurred while translating text: {e}")
```
Comment (Opoku-Agyemang) [inline src/task_queue/database_tasks/translate_contents.py:41]:
```suggestion
            log.exception(e)
```

or 

```suggestion
            log.error(message=f"Error occurred while translating text {text_hash}", exception=e)
```

File: src/translation/raw_data/campaigns_site_static_texts.json [hunk 4]
```diff
@@ -0,0 +1,349 @@
+{
```
Comment (Opoku-Agyemang) [inline src/translation/raw_data/campaigns_site_static_texts.json:1]: this should be in the frontend and there should be an endpoint which can be called to translate this in the backend and return it.

File: src/translation/translator.py [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+                elif isinstance(value, str):
+                    translated_text, err = self.translate_text(value, target_language, source_language)
+                    if err:
+                        return None, err
+                    json_data[key] = translated_text
+                else:
+                    continue
+            return json_data, None
+        except Exception as e:
+            print(f"Error translating json: {str(e)} ")
```
Comment (Opoku-Agyemang) [inline src/translation/translator.py:41]: use log.exception here as well

File: src/task_queue/database_tasks/translate_contents.py [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+                logger.log(f"Translating Footer contents for menu: {instance.name}")
+                for key, value in footer_content.items():
+                    self.translate_field(value, SOURCE_LANGUAGE_CODE)
+                    
+            return True
+        except Exception as e:
+            logger.log(f"Error occurred while translating menu model: {e}")
+            return None
+        
+    def get_valid_instances(self, model):
```
Comment (Opoku-Agyemang) [inline src/task_queue/database_tasks/translate_contents.py:134]: what is the use of this?

File: src/task_queue/database_tasks/translate_contents.py [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+                    continue
+                
+                if not (hasattr(model, 'TranslationMeta') and model.TranslationMeta.fields_to_translate):
+                    continue
+                
+                logger.info(f"\n\nTranslating contents for model: {model.__name__}\n\n")
+                
+                instances = self.get_valid_instances(model)
+                
+                for instance in instances:
```
Comment (Opoku-Agyemang) [inline src/task_queue/database_tasks/translate_contents.py:162]: why not serialize then call json translator instead?

**Review by Opoku-Agyemang — CHANGES_REQUESTED** [review]

File: src/task_queue/database_tasks/translate_contents.py (hunk 6 above)
Comment (abdullai-t) [reply src/task_queue/database_tasks/translate_contents.py:134]: I didn't want us translating stale content, which is why I added these filters.

File: src/task_queue/database_tasks/translate_contents.py (hunk 7 above)
Comment (abdullai-t) [reply src/task_queue/database_tasks/translate_contents.py:163]: The serialized version of the instances includes a lot of unnecessary data due to the joins defined in the model. This approach gives us only the data of the model being processed.

**Pull request closed without merging**